src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Circle extends Figure {
+    private double radius;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Circle(double area, String color, double radius) {
+        super(area, color);
+        this.radius = radius;
+    }
+
+    public Circle() {
+    }
+
+    public double getRadius() {
+        return radius;
+    }
+
+    @Override
+    public double countArea() {
+        return Math.PI * radius * radius;
+    }
+
+    public Circle createRandomCircle() {

```

Comment:
Circle is just data object, with `draw` and `calculateArea`. Functionality of supplying figures should be implemented in `FigureSupplier`.